python/lsst/ts/ataos/ataos_csc.py

```diff
@@ -1197,27 +1245,117 @@ async def set_hexapod(self, azimuth, elevation, axis=f'xyzuvw'):
+                    'in correction loop due to filter '
+                    'and/or disperser changes.')
+                _pointing_offsets = copy.deepcopy(self.pointing_offsets_yet_to_be_applied)
+            elif(_offset_value):
+                # No offsets above thresholds, so just return
+                self.log.debug('Focus and pointing offsets below tolerances. Passing without correcting')
+                pass
+
+            # Corrections required, so flip the bit on the detailed state
+            self.detailed_state = self.detailed_state ^ status_bit
 
-    def update_mount_position_callback(self, id_data):
-        """Callback function to update the position of the telescope.
+            self.log.debug(f'_offset_value is {_offset_value}')
+            self.log.debug(f'_pointing_offsets is {_pointing_offsets!r}')
 
-        Parameters
-        ----------
-        id_data : SALPY_ATMCS.ATMCS_mountEncoders
+            # send out even saying correction is started
+            self.evt_atspectrographCorrectionStarted.set_put(focusOffset=_offset_value,
+                                                             pointingOffsets=_pointing_offsets)
 
-        """
-        self.azimuth = id_data.azimuthCalculatedAngle[-1]
-        self.elevation = id_data.elevationCalculatedAngle[-1]
+            try:
+                # Don't need a tolerance since these values only get set if
+                # they meet a tolerance already
+                if abs(_offset_value):
+                    self.log.debug('Applying focus correction with hexapod')
+                    await self.set_hexapod(azimuth, elevation)
+
+                if np.max(abs(_pointing_offsets)):
+                    self.log.debug('Applying pointing correction _pointing_offsets'
+                                   f' = {_pointing_offsets} with atcs')
+                    # apply offsets relative to what is already there
+                    await self.atcs.offset_xy(_pointing_offsets[0], _pointing_offsets[1],
+                                              relative=True)
+                    await asyncio.sleep(0)
+                    # update accounting and remove the already applied offsets
+                    self.pointing_offsets_per_category['total'] += _pointing_offsets
+                    self.pointing_offsets_yet_to_be_applied -= _pointing_offsets
+
+                    self.log.debug("new value of pointing_offsets_per_category['total'] is"
+                                   f" {self.pointing_offsets_per_category['total']}")
+                    self.evt_pointingOffsetSummary.set_put(
+                        total=self.pointing_offsets_per_category['total'],
+                        filter=self.pointing_offsets_per_category['filter'],
+                        disperser=self.pointing_offsets_per_category['disperser'])
+
+            except Exception as e:
+                self.log.warning(f"Failed to apply spectrograph pointing offsets of {_pointing_offsets} or "
```

Comment:
I recently learned that you can do:

```
                self.log.exception(f"Failed to apply spectrograph pointing offsets of {_pointing_offsets} or "
                                  f"failed to apply focus (hexapod offset) of : {_offset_value}")
```

It will also log the traceback after the message. So, no need for the `self.log.exception(e)` afterwards